rust/lance-datafusion/src/udf/json.rs

```diff
@@ -106,18 +107,1037 @@ fn extract_json_path(jsonb_bytes: &[u8], path: &str) -> Result<Option<String>> {
+                Ok(None) => builder.append_null(),
+                Err(e) => {
+                    return Err(datafusion::error::DataFusionError::Execution(format!(
+                        "Failed to get JSON bool: {}",
+                        e
+                    )));
+                }
+            }
+        }
+    }
+
+    Ok(Arc::new(builder.finish()))
+}
+
+/// Get a field value as boolean with type coercion
+fn get_json_field_as_bool(jsonb_bytes: &[u8], key: &str) -> Result<Option<bool>> {
+    let raw_jsonb = jsonb::RawJsonb::new(jsonb_bytes);
+
+    // Try as object field first
+    let value = match raw_jsonb.get_by_name(key, false) {
+        Ok(Some(value)) => value,
+        Ok(None) => {
+            // Try as array index
+            if let Ok(index) = key.parse::<usize>() {
+                match raw_jsonb.get_by_index(index) {
+                    Ok(Some(value)) => value,
+                    Ok(None) => return Ok(None),
+                    Err(e) => {
+                        return Err(datafusion::error::DataFusionError::Execution(format!(
+                            "Failed to get array element: {}",
+                            e
+                        )))
+                    }
+                }
+            } else {
+                return Ok(None);
+            }
+        }
+        Err(e) => {
+            return Err(datafusion::error::DataFusionError::Execution(format!(
+                "Failed to get field: {}",
+                e
+            )))
+        }
+    };
+
+    // Convert to string and parse
+    let json_str = value.to_string();
+
+    // Check for null
+    if json_str == "null" {
+        return Ok(None);
+    }
+
+    // Direct boolean
+    if json_str == "true" {
+        return Ok(Some(true));
+    } else if json_str == "false" {
+        return Ok(Some(false));
+    }
+
+    // String value (remove quotes and check)
+    let s = if json_str.starts_with('"') && json_str.ends_with('"') {
+        json_str[1..json_str.len() - 1].to_lowercase()
+    } else {
+        json_str.to_lowercase()
+    };
+
+    // String to bool conversion
+    match s.as_str() {
+        "true" | "1" | "yes" | "y" | "on" => Ok(Some(true)),
+        "false" | "0" | "no" | "n" | "off" => Ok(Some(false)),
+        _ => {
+            // Try as number
+            if let Ok(n) = s.parse::<i64>() {
+                Ok(Some(n != 0))
+            } else if let Ok(f) = s.parse::<f64>() {
+                Ok(Some(f != 0.0))
+            } else if s.starts_with('[') || s.starts_with('{') {
+                Err(datafusion::error::DataFusionError::Execution(
+                    "Cannot convert JSON object or array to boolean".to_string(),
+                ))
+            } else {
+                Err(datafusion::error::DataFusionError::Execution(format!(
+                    "Cannot convert string '{}' to boolean",
+                    s
+                )))
+            }
+        }
+    }
+}
```

Comment:
These parsing functions all seem to follow a general pattern too.  I wonder if there is some reuse here.